packages/block-editor/src/components/inserter/menu.js

```diff
@@ -123,8 +125,9 @@ function InserterMenu(
 		( patternCategory, filter ) => {
 			setSelectedPatternCategory( patternCategory );
 			setPatternFilter( filter );
```

Comment:
It might be good to add an extra guard here to make sure `__experimentalOnPatternCategorySelection` is a function